concent_api/core/decorators.py

```diff
@@ -316,3 +319,20 @@ def wrapper(request: HttpRequest, golem_message: message.Message, client_public_
         response_from_view = view(request,  golem_message, client_public_key)
         return response_from_view
     return wrapper
+
+
+def retry_middleman_connection_if_not_pass_timeout(send_request_to_middleman: Callable) -> Callable:
```

Comment:
I can't see any test for this decorator. Please add some

Reply:
Did You check coverage? :D Most tests from `test_payments_sci_callback.py` go through that decorator, and the `test_that_sci_callback_should_raise_exception_on_timeout()` check every line. I can add more tests of course, but I don't know what else would I check in them?

Reply:
Yea, but these tests just go through this decorator and their are not checking the way this function works. I think that you should check it's timeouts etc. :)

Reply:
`test_that_sci_callback_should_raise_exception_on_timeout()` does. Try it with debuger. That's why I add `SCI_CALLBACK_RETRIES_TIME=1` to `override_settings`. Loop in that decorator will go on for as much seconds as You put in that setting.

Reply:
Yes, some 2-3 dedicated tests would be good.

Reply:
Oki, I tested it also and you are right. Im giving approvie, but I still think, that it should has it own test :P

Reply:
Yep, I'm trying to write that test @rwrzesien explained.

Reply:
@rwrzesien I've added 2 simple tests, check it out :)

Reply:
@Jakub89 This is not exactly what I have meant. You test if exception is being raised and raise it inside function. This does not change the main mechanism of the decorator.

Instead, you should override the settings and constant (using mock) and set it to some low values (2-3 seconds). Write three tests:
1. In first, define the dummy function which does sleep which will exceed `SCI_CALLBACK_RETRIES_TIME`. Check if exception is raised.
2. In seconds, define the dummy function which does sleep which will  **not** exceed `SCI_CALLBACK_RETRIES_TIME` and is just below this value. Return value from it. Check if value is returned.
3. In third, define the dummy function which calls a different function. That different function should make sleep and raise `SCICallbackTimeoutError` in first run, then return value which will be then returned by dummy function. The mentioned sleep does sleep should not will  **not** exceed `SCI_CALLBACK_RETRIES_TIME`. Check if value is returned, even if exception was raised and caught and function was retried.

Reply:
@rwrzesien settings are override for all tests `SCI_CALLBACK_RETRIES_TIME=1`, and I don't see why would we need to override constant?

Ad1. `test_retry_middleman_connection_returns_error_after_timout()` does exactly that, and there's no need for sleep. Dummy function raise exception all the time until one second pass and then decorator raise the same exception, which I check for.

Ad2. Doesn't `test_retry_middleman_connection_returns_correct_value()` do exactly that? It returns `None` value. And again, what's the point in `sleep`? It will only make the test last longer, with no added value. You can't manipulate that first time check with sleep, because function runs after it.

Ad3. That's how `test_that_sci_callback_should_sign_transaction_after_two_retries()` works, I tried to rewrite it similar to the decorator tests, but run in to some problems. I'll try again if it's necessary.

Reply:
Right, I thought it works a little bit different, but after examining the code I think those tests are OK.

Reply:
![tenor](https://user-images.githubusercontent.com/20682491/51472388-97690000-1d79-11e9-8eb8-603966324c7a.gif)